go/vt/vtgate/safe_session.go

```diff
@@ -282,7 +287,7 @@ func (session *SafeSession) InTransaction() bool {
 }
 
 // Find returns the transactionId and tabletAlias, if any, for a session
-func (session *SafeSession) Find(keyspace, shard string, tabletType topodatapb.TabletType) (transactionID int64, reservedID int64, alias *topodatapb.TabletAlias) {
+func (session *SafeSession) Find(keyspace, shard string, tabletType topodatapb.TabletType) (int64, int64, *topodatapb.TabletAlias, error) {
```

Comment:
I didn't find it intuitive that a method with the name `Find` actually changes the state of the shard sessions. Maybe we should rename it?